telemetry/specs/INT.mdk

```diff
@@ -624,19 +624,73 @@ corruption at preceding hops.
-3. *INT over Geneve* - Geneve is an extensible tunneling framework, allowing
+5. *INT over Geneve* - Geneve is an extensible tunneling framework, allowing
 Geneve options to be defined for INT Headers.
 
+### INT over IPv4/GRE
+
+In case the traffic being monitored is not encapsulated by any virtualization
+header, INT over VXLAN or INT over Geneve is not helpful. Instead, 
+a GRE encapsulation as defined in [RFC2784] is utilized. The INT metadata 
+header and INT metadata follows the GRE header. In an administrative domain 
+where INT is used, insertion of the INT metadata header and metadata in 
+GRE is enabled at the INT source and deletion of INT metadata header 
+and metadata is enabled at the INT sink by means of configuration. 
+
+IPv4 GRE Option format for carrying INT Header and 
+Metadata:
+
+0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+|C|R|K|S|s|Recur|   Flags | Ver |   Protocol Type = TBD_INT     |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  |
+|     Checksum (optional)       |        Offset (Optional)      |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  G
+|                         Key (Optional)                        |  R
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  E
+|                   Sequence Number (Optional)                  |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  |
+|                        Routing (Optional)                     |  |     
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+|      Type     |     Length    |        Next Protocol          |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  I
+|                                                               |  N
+|  Variable Option Data (INT Metadata Headers and Metadata]     |  T                     
+|                                                               |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+
+The GRE header and fields are defined in [RFC2784].  The GRE Protocol Type 
+value is TBD_INT.
+
+The INT Shim header for GRE option is defined as follows:
+
+* Type: This field indicates the type of INT Header following the shim header.
+Two Type values are used: one for the hop-by-hop header type and the other for
+the destination header type (See Section [#sec-int-header-types]).
+* Length: This is the total length of INT metadata header, INT stack and the
+shim header in 4-byte words. A non-INT device may read this field
+and skip over INT headers.
+* Next Protocol: 16 bits Next Protocol field contains the protocol type of the
+packet follwing INT Metadata Header and Metadata. When the most significant 
+octet is 0x00, the Protocol Type is taken to be an IP Protocol Number as 
```

Comment:
The option to throw in an IP protocol number was removed between draft-weis-ippm-ioam-eth-00 and draft-weis-ippm-ioam-eth-01. I don't remember the history behind this. Is it necessary?

Reply:
Checked with IOAM folks. Protocol number was removed to be consistent with Geneve. I am fine with Ethertype. Corrected the text